### PR TITLE
[IE CLDNN] Add b_fs_yx_fsv16 concat optimizations

### PR DESCRIPTION
**Pull request by kdobros**
1. Add fsv16 int8 support to optimized kernel
2. Optimize fsv16 concat kernel
3. Add graph optimization to improve concat alignment

Issue: CVS-28494

### CODE REVIEW

**Comment by sshlyapn** [comment]
Looks good to me!

File: inference-engine/thirdparty/clDNN/src/graph_optimizer/concat_input_order.cpp [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+
+    return false;
+}
+
+void shuffle_weights(data_node& node, const std::vector<shuffle_range>& ranges) {
+    // Correct for shuffled features by shuffling input feature dimension in weights.
+    // This allows to restore correct feature order on output and only changes calculation order.
+    auto wei_layout = node.get_output_layout();
+    auto& old_weights_memory = node.get_attached_memory();
+    auto new_weights_memory = old_weights_memory.get_engine()->allocate_memory(wei_layout, old_weights_memory.get_net_id(), true);
```
Comment (vladimir-paramuzov) [inline inference-engine/thirdparty/clDNN/src/graph_optimizer/concat_input_order.cpp:73]: Do you really need to reset this buffer? The layouts are equal, so I'd expect that all elements are initialized in the loop below

File: inference-engine/thirdparty/clDNN/src/graph_optimizer/concat_input_order.cpp [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+using shuffle_range = std::pair<int32_t, int32_t>;
+
+bool can_shuffle_features(program_node& node) {
+    if (node.is_type<convolution>()) {
+        auto& conv_node = node.as<convolution>();
+        auto& wei_node = conv_node.weights();
+
+        return conv_node.get_groups() == 1 && conv_node.get_split() == 1 &&
+            conv_node.get_deformable_groups() == 1 && !conv_node.get_transposed() &&
+            !conv_node.activations_zero_points_term() &&
```
Comment (vladimir-paramuzov) [inline inference-engine/thirdparty/clDNN/src/graph_optimizer/concat_input_order.cpp:41]: What about weights zero points?

Comment (kdobros) [reply inference-engine/thirdparty/clDNN/src/graph_optimizer/concat_input_order.cpp:41]: Aren't they always per output feature? So changing input features order shouldn't affect that.

Comment (vladimir-paramuzov) [reply inference-engine/thirdparty/clDNN/src/graph_optimizer/concat_input_order.cpp:41]: OK, you're right

File: inference-engine/thirdparty/clDNN/src/graph_optimizer/concat_input_order.cpp (hunk 1 above)
Comment (kdobros) [reply inference-engine/thirdparty/clDNN/src/graph_optimizer/concat_input_order.cpp]: Changed to only reset when there is padding or layout is blocked

**Pull request merged**